### PR TITLE
fix(gatsby): Fix issue with env variables not being passed to getServerData

### PR DESCRIPTION
**Pull request by marvinjude**
## Description

This PR addresses an issue where env vars aren't passed into [`getServerData`](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/#step-1-adding-getserverdata-function) by passing env vars via webpack as opposed to using `babel-plugin-env-vars`. This helps us escape a caching problem that causes this issue.

This reverts #33690

## Related Issues
Fixes [sc-42081]

### CODE REVIEW

File: e2e-tests/production-runtime/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
   ],
   "license": "MIT",
   "scripts": {
-    "build": "cross-env CYPRESS_SUPPORT=y gatsby build",
+    "build": "cross-env FROM_COMMAND_LINE=YES CYPRESS_SUPPORT=y gatsby build",
```
Comment (pieh) [inline e2e-tests/production-runtime/package.json:33]: Consider adding env var with `GATSBY_` prefix to make sure it's available both for browser bundle and for getServerData

File: packages/gatsby/src/commands/build-html.ts [hunk 2]
```diff
@@ -125,6 +125,11 @@ const runWebpack = (
     } else {
       build(compilerConfig).then(
         ({ stats, close }) => {
+          fs.outputJsonSync(
+            `${directory}/${ROUTES_DIRECTORY}/.env`,
+            process.env
+          )
+
```
Comment (pieh) [inline packages/gatsby/src/commands/build-html.ts:132]: need to revert this change

File: e2e-tests/production-runtime/package.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
   ],
   "license": "MIT",
   "scripts": {
-    "build": "cross-env CYPRESS_SUPPORT=y gatsby build",
+    "build": "cross-env GATSBY_PREFIXED_FROM_COMMAND_LINE=YES FROM_COMMAND_LINE=YES CYPRESS_SUPPORT=y gatsby build",
     "build:offline": "cross-env TEST_PLUGIN_OFFLINE=y CYPRESS_SUPPORT=y gatsby build",
```
Comment (pieh) [inline e2e-tests/production-runtime/package.json:34]: I think we need to add env vars to `build:offline` script variant as well. We run couple different builds in our tests - first regular one and run tests on them and after that we do `build:offline` and run tests again.

Missing env vars in one of them will case our tests to fail

```suggestion
    "build:offline": "cross-env GATSBY_PREFIXED_FROM_COMMAND_LINE=YES FROM_COMMAND_LINE=YES TEST_PLUGIN_OFFLINE=y CYPRESS_SUPPORT=y gatsby build",
```

Comment (marvinjude) [reply e2e-tests/production-runtime/package.json:34]: Ah! I see. Thanks for spotting that!

**Review by LekoArts — APPROVED** [review]

**Comment by marvinjude** [comment]
Published in  `gatsby@4.5.3`
